pkg/policy/policy.go

```diff
@@ -78,19 +77,41 @@ func (s *SearchContext) String() string {
 	for _, toLabel := range s.To {
 		to = append(to, toLabel.String())
 	}
+	var str strings.Builder
```

Comment:
nitpicking: Can we have a comment here discouraging attempts to "simplify" this codepath with `fmt.Sprintf()` explaining the rational behind this patch implementation (and/or linking to https://github.com/cilium/cilium/issues/19571) please?